src/org/mozilla/javascript/Undefined.java

```diff
@@ -37,26 +41,106 @@ public boolean equals(Object obj) {
     @Override
     public int hashCode() {
         // All instances of Undefined are equivalent!
-        return 0;
+        return System.identityHashCode(instance);
```

Comment:
because instance is final maybe we can make this a constant also

Reply:
Good idea. Thanks!